tests/test_html_files.py

```
@@ -0,0 +1,66 @@
+    assert 'head' in parser.tags or 'body' in parser.tags, f"Missing head/body tag in {filepath}"
+
+def test_index_html_redirect():
+    index_path = os.path.join(ROOT_DIR, 'index.html')
+    with open(index_path, 'r', encoding='utf-8') as f:
+        content = f.read()
+    assert 'Web%20Ui/login.html' in content or 'login.html' in content
+
+def test_web_ui_critical_elements():
+    main_html_path = os.path.join(ROOT_DIR, 'Web Ui', 'main.html')
+    if os.path.exists(main_html_path):
+        with open(main_html_path, 'r', encoding='utf-8') as f:
```

Ruff (0.16.2)

[warning] Unnecessary mode argument

Remove mode argument

(UP015)